src/bmi_pet.c

```diff
@@ -1015,6 +1059,15 @@ static int Get_value(Bmi * self, const char * name, void *dest)
 
 static int Set_value (Bmi *self, const char *name, void *array)
 {
+    // special cases for serialization
```

Comment:
Again, this feels brittle, and there isn't any documentation on how these special cases are supposed to work and be used.